pkg/machinepool/machinepool.go

```diff
@@ -35,6 +82,1061 @@ func NewMachinePoolService() MachinePoolService {
+	if err != nil {
+		return fmt.Errorf("Failed to add machine pool to cluster '%s': %v", clusterKey, err)
+	}
+
+	if output.HasFlag() {
+		if err = output.Print(createdMachinePool); err != nil {
+			return fmt.Errorf("Unable to print machine pool: %v", err)
+		}
+	} else {
+		r.Reporter.Infof("Machine pool '%s' created successfully on cluster '%s'", name, clusterKey)
+		r.Reporter.Infof("To view the machine pool details, run 'rosa describe machinepool --cluster %s --machinepool %s'",
+			clusterKey, name)
+		r.Reporter.Infof("To view all machine pools, run 'rosa list machinepools --cluster %s'", clusterKey)
+	}
+
+	return nil
+}
+
+func Split(r rune) bool {
+	return r == '=' || r == ':'
+}
+
+// getMachinePoolAvailabilityZones derives the availability zone from the user input or the cluster spec
+func getMachinePoolAvailabilityZones(r *rosa.Runtime, cluster *cmv1.Cluster, multiAZMachinePool bool,
+	availabilityZoneUserInput string, subnetUserInput string) ([]string, error) {
+	// Single AZ machine pool for a multi-AZ cluster
+	if cluster.MultiAZ() && !multiAZMachinePool && availabilityZoneUserInput != "" {
+		return []string{availabilityZoneUserInput}, nil
+	}
+
+	// Single AZ machine pool for a BYOVPC cluster
+	if subnetUserInput != "" {
+		availabilityZone, err := r.AWSClient.GetSubnetAvailabilityZone(subnetUserInput)
+		if err != nil {
+			return []string{}, err
+		}
+
+		return []string{availabilityZone}, nil
+	}
+
+	// Default option of cluster's nodes availability zones
+	return cluster.Nodes().AvailabilityZones(), nil
+}
+
+func minReplicaValidator(multiAZMachinePool bool) interactive.Validator {
+	return func(val interface{}) error {
+		minReplicas, err := strconv.Atoi(fmt.Sprintf("%v", val))
+		if err != nil {
+			return err
+		}
+		if minReplicas < 0 {
+			return fmt.Errorf("min-replicas must be a non-negative integer")
+		}
+		if multiAZMachinePool && minReplicas%3 != 0 {
+			return fmt.Errorf("Multi AZ clusters require that the replicas be a multiple of 3")
+		}
+		return nil
+	}
+}
+
+func maxReplicaValidator(minReplicas int, multiAZMachinePool bool) interactive.Validator {
+	return func(val interface{}) error {
+		maxReplicas, err := strconv.Atoi(fmt.Sprintf("%v", val))
+		if err != nil {
+			return err
+		}
+		if minReplicas > maxReplicas {
+			return fmt.Errorf("max-replicas must be greater or equal to min-replicas")
+		}
+		if multiAZMachinePool && maxReplicas%3 != 0 {
+			return fmt.Errorf("Multi AZ clusters require that the replicas be a multiple of 3")
+		}
+		return nil
+	}
+}
+
+func spotMaxPriceValidator(val interface{}) error {
+	spotMaxPrice := fmt.Sprintf("%v", val)
+	if spotMaxPrice == "on-demand" {
+		return nil
+	}
+	price, err := strconv.ParseFloat(spotMaxPrice, commonUtils.MaxByteSize)
+	if err != nil {
+		return fmt.Errorf("Expected a numeric value for spot max price")
+	}
+
+	if price <= 0 {
+		return fmt.Errorf("Spot max price must be positive")
+	}
+	return nil
+}
```

Comment:
NP : might be worth breaking these out into their own file, try bundle the machine pool validations together